### PR TITLE
ABTest, Plans: Launch `jetpackNewPurchaseIntroPricing` test

### PR DESCRIPTION
**Pull request by mattgawarecki**
#### Changes proposed in this Pull Request

Launches `1196108640073826-as-1199527862646257`.

* Create a new test in `active-tests.js` called `jetpackNewPurchaseIntroPricing` with two variants, `withIntroPricing` and `withoutIntroPricing`.
* Create a new iteration code-named `NPIP`, which stands for "new purchase introductory pricing."
* Add logic to activate the NPIP iteration when the test variant reported for the current user is `withIntroPricing`.
* Create a new set of product card and product grid components for the NPIP iteration, copying originally from the default (I5) iteration.
* Modify styling and layout for the **Plans** and **Pricing** pages in the NPIP iteration to match the mock-ups in `zp0udhbEB9dvdS4rB3GJ0z-fi-923%3A0` (see diagram i.8).

#### Implementation notes

I'm aware that a lot of the component code is copied and pasted. I did this for two reasons:
1. To save time developing the test; and
2. To make the test code easier to remove in the future.

It's not the cleanest way to do things -- in fact it makes me a bit itchy -- but it was a compromise I made in the name of delivering on time.

#### Testing instructions

**TIP**: Compare closely with `1196108640073826-as-1199527862646257` for original design references. It includes links to mock-ups and a video walkthrough in the description.

##### Setup

* Open Calypso Blue in your testing environment.
* In the development helper pane in the lower right of the screen, hover over ABTESTS and then find `jetpackNewPurchaseIntroPricing`. Testing instructions follow for the `withIntroPricing` variant. The `withoutIntroPricing` variant should render all pages as they currently exist in production.
* Alternatively, or for environments in which the development helper isn't available, you can append `cloud-pricing-page=npip` as a query parameter in your browser to show the NPIP variant.

##### Intro pricing banner

* Verify that an introductory pricing banner is visible at the top of the page, mentioning our 14-day money back guarantee.

image image
image
image

##### *(Calypso Green ONLY)* Featured product header color

* Verify that the "Recommended" header and border in Calypso Green now use `studio-pink-50` as their color (i.e., `#C9356E`). (NOTE: In other environments, this header should match the active theme, as in production.)

image

##### "You own this" and "Part of your current plan" markers

* Verify for products or plans that are already owned or included in an existing site purchase, a marker showing "You own this" or "Part of your current plan" is displayed in place of the item's price. The marker should be on a light-gray background and be accompanied by a check-mark icon.

image image

##### Scratched-out prices

* Verify that all products with discounts show the original price "scratched out" with a pink horizontal line, followed by the discounted price.
* Verify that all discounted prices are shown as if a 40% off coupon were already applied (e.g., if the original price was $19.95 per month, the discounted price should be $11.97 per month).

image

##### Billing term copy

* Verify that billing terms are written as either "/month, paid yearly" or "/month, paid monthly," depending on which term is currently selected.

image image

##### "You save 40%" and "Get started for free" ribbons

* Verify that all non-free, non-owned products show a yellow trapezoidal ribbon under their price with the bolded label "* You Save 40%."

image 

##### Positioning of product descriptions

* Verify that product descriptions have been moved from under the product name to below the call-to-action button.

image

##### Feature list bullet points

* Verify that the feature bullet points next to all products are now standard "discs" instead of green check marks.

image image

##### Mid-page asterisk clauses

* Verify that in the middle of the page, where "All plans include priority support" is displayed, two more asterisk clauses are displayed above it:
  * Special introductory pricing, all renewals are at full price. 
  * Monthly plans are 7-day money back guarantee.

image

### CODE REVIEW

File: client/components/jetpack/card/i5/jetpack-product-card-i5/index.tsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { Button, ProductIcon } from '@automattic/components';
 /**
  * Internal dependencies
  */
+import { preventWidows } from 'calypso/lib/formatting';
```
Comment (mattgawarecki) [inline client/components/jetpack/card/i5/jetpack-product-card-i5/index.tsx:13]: This is an unrelated one-line bugfix. We call `preventWidows` on line 202.

File: client/components/jetpack/card/npip/jetpack-free-card-npip/index.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/**
```
Comment (mattgawarecki) [inline client/components/jetpack/card/npip/jetpack-free-card-npip/index.tsx:1]: Recreated from i5 specifically to accommodate styling and slight markup differences re: feature list bullet points.

File: client/components/jetpack/card/npip/jetpack-product-card-npip/features-item.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/**
```
Comment (mattgawarecki) [inline client/components/jetpack/card/npip/jetpack-product-card-npip/features-item.tsx:1]: Copied from i5 to address feature list bullet points.

File: client/components/jetpack/card/npip/jetpack-product-card-npip/index.tsx [hunk 4]
```diff
@@ -0,0 +1,269 @@
+/**
```
Comment (mattgawarecki) [inline client/components/jetpack/card/npip/jetpack-product-card-npip/index.tsx:1]: Lots of important changes here re: the following:
* "You own this"
* "* You Save 40%" / "Get started for free"
* Positioning of product description

File: client/components/jetpack/card/npip/jetpack-product-card-npip/style.scss [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+	@include breakpoint-deprecated( '>660px' ) {
+		min-height: 108px;
+	}
+
+	.is-disabled & {
+		opacity: 0.4;
+	}
+}
+
+.jetpack-product-card-npip__you-own-this {
```
Comment (mattgawarecki) [inline client/components/jetpack/card/npip/jetpack-product-card-npip/style.scss:104]: Creates the light gray background for "you own this"

File: client/components/jetpack/card/npip/jetpack-product-card-npip/style.scss [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+		opacity: 0.4;
+	}
+}
+
+.jetpack-product-card-npip__price
+.jetpack-product-card-npip__original-price {
+	margin-inline-end: 16px;
+}
+
+.jetpack-product-card-npip__you-save,
```
Comment (mattgawarecki) [inline client/components/jetpack/card/npip/jetpack-product-card-npip/style.scss:138]: The "You Save 40%" and "Get started for free" ribbons are styled here.

File: client/components/jetpack/card/npip/jetpack-product-card-npip/style.scss [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+	display: inline-flex;
+
+	color: var( --studio-gray-100 );
+}
+
+.jetpack-product-card-npip__price
+.plan-price.is-original {
+	color: var( --studio-gray-20 );
+
+	&::before {
```
Comment (mattgawarecki) [inline client/components/jetpack/card/npip/jetpack-product-card-npip/style.scss:185]: This is the pink horizontal price scratch-out.

File: client/components/jetpack/intro-pricing-banner/index.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
+/**
```
Comment (mattgawarecki) [inline client/components/jetpack/intro-pricing-banner/index.tsx:1]: Copied from the Fresh Start sale banner.

File: client/lib/abtest/active-tests.js [hunk 9]
```diff
@@ -68,4 +68,13 @@ export default {
 		defaultVariation: 'control',
 		allowExistingUsers: false,
 	},
+	jetpackNewPurchaseIntroPricing: {
```
Comment (mattgawarecki) [inline client/lib/abtest/active-tests.js:71]: This is the A/B test setup.

**Review by mattgawarecki — COMMENTED** [review]
Leaving some helpful guideposts for reviewers.

**Comment by eeeeevon13** [comment]
@mattgawarecki  Looks good. Can you make the the *text ( all plans include priority support etc) not bolded, just regular weight?

**Comment by mattgawarecki** [comment]
> Can you make the the *text ( all plans include priority support etc) not bolded, just regular weight?

@eeeeevon13 Inspecting in the browser, I'm seeing that the items you mention are already set to a font weight of 400, so it seems they're not actually bolded? Looking at Google Fonts, I'm not seeing any lighter-weight options ready to go, either. 🙁 I could always upload another specific font variant, but that might get a little burdensome for people on low-bandwidth connections since we've already added SemiCondensed Bold recently (a 120KB payload). We could also find an alternative font, or punt as a last resort and just leave the asterisk items the way they are for now.

What do you think our best path forward might be?

**Comment by rcanepa** [comment]
Not sure how much of a blocker is this, but on Safari Site Search card looks like this:


Every other card looks fine in that browser.

**Update**

This is not an issue with Safari. It happens when Site Search price is too long (e.g. in Brazil's currency) and it wraps to the next line.

**Review by rcanepa — APPROVED** [review]
Besides the small UI issue with Site Search, everything tested very well on every browser I tried and on different viewport sizes.

Code changes look good and duplication concerns will be solved once we merge the PRs that remove previous iterations.

Very nice and clean implementation. Awesome job, Matt!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5531649

Thank you @mattgawarecki for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5531649).
